examples/vision/handwriting_recognition.py

```diff
@@ -338,91 +349,95 @@ def prepare_dataset(image_paths, labels):
 """
 ## Model
 
-Our model will use the CTC loss as an endpoint layer. For a detailed understanding of the
-CTC loss, refer to [this post](https://distill.pub/2017/ctc/).
+Our model will use the CTC loss. For a detailed understanding of the CTC loss,
+refer to [this post](https://distill.pub/2017/ctc/).
 """
 
 
-class CTCLayer(keras.layers.Layer):
-    def __init__(self, name=None):
-        super().__init__(name=name)
-        self.loss_fn = keras.backend.ctc_batch_cost
-
-    def call(self, y_true, y_pred):
-        batch_len = tf.cast(tf.shape(y_true)[0], dtype="int64")
-        input_length = tf.cast(tf.shape(y_pred)[1], dtype="int64")
-        label_length = tf.cast(tf.shape(y_true)[1], dtype="int64")
-
-        input_length = input_length * tf.ones(shape=(batch_len, 1), dtype="int64")
-        label_length = label_length * tf.ones(shape=(batch_len, 1), dtype="int64")
-        loss = self.loss_fn(y_true, y_pred, input_length, label_length)
-        self.add_loss(loss)
-
-        # At test time, just return the computed predictions.
-        return y_pred
+# Ported from keras 2 backend (with keras 3 ops updates).
+def ctc_decode(y_pred, input_length, greedy=True, beam_width=100, top_paths=1):
```

Comment:
In the future you should be able to use ops from https://github.com/keras-team/keras/pull/19366